### ISSUE TITLE
az upgrade under Windows waits for end of msiexec which wants to update python.exe

### ISSUE THREAD

**Issue by lgiuliani80** [Azure/azure-cli#23947]
**Related command**
az upgrade

**Describe the bug**
az upgrade under Windows downloads and runs the MSI installer. The call to msiexec.exe is synchronous (i.e. it waits for its termination), but during the installation process the MSI wants to update python.exe, which is in use.

**To Reproduce**
az upgrade [under Windows]

**Expected behavior**
Under Windows EXIT PROCESS after calling msiexec: the upgrade will involve UI, so there is no need to keep the console blocked. This way this bug is prevented.

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by miqm** [comment]
👍🏻 same here.
I assume it's because Azure CLI is waiting for installer to finish - so the python.exe is still running. So either az cli installer should not require python to exit or the cli after starting installer should exit and hope that user will complete installation.

**Comment by stijnherreman** [comment]
The previous time I tried upgrading this recent install, I tried letting the installer shut down the Python process but it failed to do so. So I continued with the reboot option.
This time I tried to stop the process with Ctrl-C, but the process was unresponsive. Then I killed the Python process through Task Manager, and only then "Terminate batch job (Y/N)?" appeared.

I'd expect anyone on Windows running `az upgrade` is running into this issue, including the people who developed/tested this feature. Does this only occur under a specific environment? I'm using PowerShell Core in Windows Terminal.

---

> cli after starting installer should exit and hope that user will complete installation

@miqm Does it matter to the CLI whether or not the user completes the installation?
